public/app/features/dashboard-scene/sharing/ExportButton/ExportAsJson.tsx

```diff
@@ -0,0 +1,106 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import { useAsync } from 'react-use';
+import AutoSizer from 'react-virtualized-auto-sizer';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { selectors as e2eSelectors } from '@grafana/e2e-selectors';
+import { SceneComponentProps } from '@grafana/scenes';
+import { Button, ClipboardButton, CodeEditor, Label, Spinner, Stack, Switch, useTheme2 } from '@grafana/ui';
+import { Trans } from 'app/core/internationalization';
+
+import { ShareExportTab } from '../ShareExportTab';
+
+const selector = e2eSelectors.pages.ExportDashboardDrawer.ExportAsJson;
+
+export class ExportAsJson extends ShareExportTab {
+  static Component = ExportAsJsonRenderer;
+}
+
+function ExportAsJsonRenderer({ model }: SceneComponentProps<ExportAsJson>) {
+  const theme = useTheme2();
+  const styles = getStyles(theme);
+
+  const { isSharingExternally, dashboardRef } = model.useState();
+
+  const onCancelClick = () => {
+    dashboardRef.resolve().closeModal();
+  };
+
+  const dashboardJson = useAsync(async () => {
+    const json = await model.getExportableDashboardJson();
+    return JSON.stringify(json, null, 2);
+  }, [isSharingExternally]);
+
+  return (
+    <>
+      <p>
+        <Trans i18nKey="export.json.info-text">
+          Copy or download a JSON file containing the JSON of your dashboard.
+        </Trans>
+      </p>
+
+      <Stack gap={1} alignItems="center">
+        <Switch
+          data-testid={selector.exportExternallyToggle}
+          id="export-externally-toggle"
+          value={isSharingExternally}
+          onChange={model.onShareExternallyChange}
+        />
+        <Label>
+          <Trans i18nKey="export.json.export-externally-label">Export the dashboard to use in another instance</Trans>
+        </Label>
+      </Stack>
+
+      <AutoSizer disableHeight className={styles.codeEditorBox} data-testid={selector.codeEditor}>
+        {({ width }) => {
+          if (dashboardJson.value) {
+            return (
+              <CodeEditor
+                value={dashboardJson.value}
+                language="json"
+                showMiniMap={false}
+                height="500px"
+                width={width}
+                readOnly={true}
+              />
+            );
+          }
+
+          return dashboardJson.loading && <Spinner />;
+        }}
+      </AutoSizer>
+
+      <Stack gap={1} flex={1} direction={{ xs: 'column', sm: 'row' }}>
```

Comment:
great! Take a look at ShareExternally that it also adds a paddingBottom

Reply:
why do we need paddingBottom? There is nothing after the buttons

Reply:
it adds space after scrolling in responsive view